lms/djangoapps/courseware/date_summary.py

```diff
@@ -130,7 +148,7 @@ class CourseEndDate(DateSummary):
 
     @property
     def description(self):
-        if datetime.now(pytz.UTC) <= self.date:
+        if self.date is None or datetime.now(pytz.UTC) <= self.date:
```

Comment:
You're right, and now that I think about it it doesn't really even make sense to enable the end date block if there's no end date.